shelf_router/lib/src/router.dart

```diff
@@ -147,9 +147,8 @@ class Router {
   ///
   /// In this case prefix may not contain any parameters, nor
   void mount(String prefix, Handler handler) {
-    if (!prefix.startsWith('/') || !prefix.endsWith('/')) {
-      throw ArgumentError.value(
-          prefix, 'prefix', 'must start and end with a slash');
+    if (!prefix.startsWith('/')) {
+      throw ArgumentError.value(prefix, 'prefix', 'must start with a slash');
     }
 
     // first slash is always in request.handlerPath
```

Comment:
Hmm, maybe we need to do:

```dart
    final path = prefix.substring(1);
    if (prefix.endsWith('/')) {
      all(prefix + '<path|[^]*>', (Request request) {
        return handler(request.change(path: path));
      });
    } else {
      all(prefix, (Request request) {
        return handler(request.change(path: path));
      });
      all(prefix + '/<path|[^]*>', (Request request) {
        return handler(request.change(path: path + '/'));
      });
    }
```

Aren't we really trying to handle two cases here? The one with a trailing slash and the one without a trailing slash.
And if we mount without a trailing slash, then we have two patterns, `/prefix` and `/prefix/<whatever>`.

I do wonder if `request.change(path: 'prefix')` will actually work with a nested router that's trying to match `/`, but that's probably something we could make a test case for.

Reply:
> It still looks like that the 'api2' call is directed to the 'api' handler. I was not able to find the root cause yet.

Yeah, that because when prefix is `/api` then your pattern becomes `/api<path|[^]*>` which is probably not what you want. As a request for `/api2` will match pattern `/api<path|[^]*>`.

Hence, I propose that we make two patterns as outlined above.